ReleaseTooling/Sources/PrivacyKit/AccessedAPIType.swift

```diff
@@ -0,0 +1,128 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+
+/// A structure containing metadata about the usage of API that requires justification for use.
+/// https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_use_of_required_reason_api
+public struct AccessedAPIType: Encodable {
+  public enum Category: String, Encodable {
+    case fileTimestamp = "NSPrivacyAccessedAPICategoryFileTimestamp"
+    case systemBootTime = "NSPrivacyAccessedAPICategorySystemBootTime"
+    case diskSpace = "NSPrivacyAccessedAPICategoryDiskSpace"
+    case activeKeyboards = "NSPrivacyAccessedAPICategoryActiveKeyboards"
+    case userDefaults = "NSPrivacyAccessedAPICategoryUserDefaults"
+
+    /// The possible reasons that this category of API may be accessed.
+    public var possibleReasons: [Reason] {
+      // TODO(ncooke3): It would be nice if reasons were scoped to categories
+      // in a type-safe way.
+      switch self {
+      case .fileTimestamp: return [._DDA9_1, ._C617_1, ._3B52_1]
+      case .systemBootTime: return [._35F9_1]
+      case .diskSpace: return [._85F4_1, ._E174_1]
+      case .activeKeyboards: return [._3EC4_1, ._54BD_1]
+      case .userDefaults: return [._CA92_1]
+      }
+    }
+  }
+
+  public enum Reason: String, Encodable {
+    // File timestamp APIs
+    case _DDA9_1 = "DDA9.1"
+    case _C617_1 = "C617.1"
+    case _3B52_1 = "3B52.1"
```

Comment:
These reason codes come from Apple's docs. Because enum cases cannot start with a number, I'm prefixing the cases with underscores. All cases are prefixed for consistency.